duckduckgo/macos-browser
# Prevent crash on database init failure

Task/Issue URL: https://app.asana.com/0/1199230911884351/1205479931554357/f

**Description**:
- If Database initialization fails for some reason we‘re normally showing an alert and terminating; 
- But since the Database is initialized early in the app launch process showing the alert leads to the crash because of uninitialized dependencies in `applicationDidActivate`

**Steps to test this PR**:
1. Add `NSAlert().runModal()` early in `AppDelegate.applicationWillFinishLaunching`
2. Run the app, validate the alert is shown without the crash and the app can be activated/deactivated while it‘s shown



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
